src/core/etl/src/Flow/ETL/PHP/Type/Logical/List/ListElement.php

```diff
@@ -47,14 +47,14 @@ public static function map(MapType $type) : self
     /**
      * @param class-string $class
      */
-    public static function object(string $class) : self
+    public static function object(string $class, bool $optional = false) : self
```

Comment:
I think `nullable` is a better name, since this flag is not making list element optional, it allows it to be a null but it still needs to be present.